src/utils/components/DiskModal/components/DiskSourceSelect/components/DiskSourceClonePVCSelect/DiskSourceClonePVCSelectName.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, useMemo } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+
+import { bytesToDiskSize } from '@catalog/utils/quantity';
+import {
+  modelToGroupVersionKind,
+  PersistentVolumeClaimModel,
+} from '@kubevirt-ui/kubevirt-api/console';
+import { DiskFormState } from '@kubevirt-utils/components/DiskModal/utils/types';
+import InlineFilterSelect from '@kubevirt-utils/components/FilterSelect/InlineFilterSelect';
+import FormGroupHelperText from '@kubevirt-utils/components/FormGroupHelperText/FormGroupHelperText';
+import Loading from '@kubevirt-utils/components/Loading/Loading';
+import { useKubevirtTranslation } from '@kubevirt-utils/hooks/useKubevirtTranslation';
+import usePVCs from '@kubevirt-utils/hooks/usePVCs';
+import { convertResourceArrayToMap, getName } from '@kubevirt-utils/resources/shared';
+import { isEmpty } from '@kubevirt-utils/utils/utils';
+import { FormGroup, ValidatedOptions } from '@patternfly/react-core';
+
+const DiskSourceClonePVCSelectName: FC = () => {
+  const { t } = useKubevirtTranslation();
+  const {
+    control,
+    formState: { errors },
+    setValue,
+    watch,
+  } = useFormContext<DiskFormState>();
+
+  const namespace = watch('pvc.pvcNamespace');
+
+  const [pvcs, pvcsLoaded] = usePVCs(namespace);
+
+  const pvcNames = useMemo(() => pvcs?.map(getName), [pvcs]);
+
+  const pvcMapper = convertResourceArrayToMap(pvcs);
+  if (!pvcsLoaded) return <Loading />;
+
+  const fieldId = 'pvc-name-select';
```

Comment:
i think better to a united const file for all fields ids, there are quite a few across this pr